diffsync/store/redis.py

```diff
@@ -0,0 +1,242 @@
+"""RedisStore module."""
+import copy
+import uuid
+from pickle import loads, dumps  # nosec
+from typing import List, Mapping, Text, Type, Union, TYPE_CHECKING
+
+from redis import Redis
+from redis.exceptions import ConnectionError as RedisConnectionError
+from diffsync.exceptions import ObjectNotFound, ObjectStoreException, ObjectAlreadyExists
+from diffsync.store import BaseStore
+
+if TYPE_CHECKING:
+    from diffsync import DiffSyncModel
+
+REDIS_DIFFSYNC_ROOT_LABEL = "diffsync"
+
+
+class RedisStore(BaseStore):
+    """RedisStore class."""
+
+    def __init__(self, *args, store_id=None, host="localhost", port=6379, url=None, db=0, **kwargs):
+        """Init method for RedisStore."""
+        super().__init__(*args, **kwargs)
+
+        try:
+            if url:
+                self._store = Redis.from_url(url, db=db)
+            else:
+                self._store = Redis(host=host, port=port, db=db)
```

Comment:
I'm not a big fan of having both `url` and `host/port` as mutually exclusive arguments. Can we converge to just one or the other? If not, can we have some error checking to enforce that they actually are mutually exclusive?

Reply:
The url was added to be able to use the test emulation library for Redis. I will enforce the checking.